code/modules/power/smes.dm

```diff
@@ -221,18 +227,60 @@
 
 /obj/machinery/power/smes/update_overlays()
 	. = ..()
-	if(machine_stat & BROKEN)
-		return
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
 
 	if(panel_open)
+		. += "panel"
 		return
 
-	. += "smes-op[outputting ? 1 : 0]"
-	. += "smes-oc[inputting ? 1 : 0]"
+	if(machine_stat & BROKEN)
+		return
 
 	var/clevel = chargedisplay()
-	if(clevel > 0)
-		. += "smes-og[clevel]"
+	if(!terminal)
+		. += "failing"
+		SSvis_overlays.add_vis_overlay(src, icon, "failing", layer, plane, dir)
+		SSvis_overlays.add_vis_overlay(src, icon, "failing", layer, EMISSIVE_PLANE, dir)
+	if(clevel>0)
+		. += "charge[clevel]"
+		SSvis_overlays.add_vis_overlay(src, icon, "charge[clevel]", layer, plane, dir)
+		SSvis_overlays.add_vis_overlay(src, icon, "charge[clevel]", layer, EMISSIVE_PLANE, dir)
+	if(is_emped)
+		. += "emp"
+		SSvis_overlays.add_vis_overlay(src, icon, "emp", layer, plane, dir)
+		SSvis_overlays.add_vis_overlay(src, icon, "emp", layer, EMISSIVE_PLANE, dir)
+	else
+		if(inputting)
+			if(clevel == SMES_CLEVEL_5)
+				. += "input-2"
+				SSvis_overlays.add_vis_overlay(src, icon, "input-2", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "input-2", layer, EMISSIVE_PLANE, dir)
+			else
+				. += "input-1"
+				SSvis_overlays.add_vis_overlay(src, icon, "input-1", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "input-1", layer, EMISSIVE_PLANE, dir)
+		else if(input_attempt)
+			. += "input-0"
+			SSvis_overlays.add_vis_overlay(src, icon, "input-0", layer, plane, dir)
+			SSvis_overlays.add_vis_overlay(src, icon, "input-0", layer, EMISSIVE_PLANE, dir)
+		else
+			. += "input-off"
+			SSvis_overlays.add_vis_overlay(src, icon, "input-off", layer, plane, dir)
+			SSvis_overlays.add_vis_overlay(src, icon, "input-off", layer, EMISSIVE_PLANE, dir)
+
+		if(outputting)
+			if(clevel == SMES_CLEVEL_5)
+				. += "output2"
+				SSvis_overlays.add_vis_overlay(src, icon, "output2", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "output2", layer, EMISSIVE_PLANE, dir)
+			else
+				. += "output1"
+				SSvis_overlays.add_vis_overlay(src, icon, "output1", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "output1", layer, EMISSIVE_PLANE, dir)
+		else
+			. += "output0"
+			SSvis_overlays.add_vis_overlay(src, icon, "output0", layer, plane, dir)
+			SSvis_overlays.add_vis_overlay(src, icon, "output0", layer, EMISSIVE_PLANE, dir)
```

Comment:
This seems a little dumb. Might be best to work out the clevel number first, and then reuse it wherever it is called for.
Also of note, why do the input icon states separate with a - and the output ones don't? that seems dumb

Reply:
Tried to redo using if(inputting && clevel == SMES_CLEVEL_5) but it doesnt seem to work as intended after testing it. Also for the other thing, it was just typos under dmi states names, fixed them to have output-